### ISSUE TITLE
Calico pods in newly launched HA microk8s cluster stuck in CrashLoopBackoff

### ISSUE THREAD

**Issue by devec0** [ubuntu/microk8s#1545]
Ubuntu 20.04 LTS (ARM64)
Microk8s 1.19 (1669)
Kernel 5.6.0-1137-ayufan-ge57f05e7bf8f

When launching a new microk8s cluster on the 1.19 stable channel of microk8s, Calico does not enter a ready state when adding three HA nodes. Testing previously on the `latest/edge/ha-preview` channel did not show this same behaviour on the same systems. I have removed microk8s and reinstalled it freshly, bootstrapping a new 3 node cluster each time, and have been able to reproduce this behaviour.

node1,2 & 3: `sudo snap install microk8s --channel 1.19/stable --classic`

node1,2,3: -
```
microk8s status
microk8s is running
high-availability: no
  datastore master nodes: 127.0.0.1:19001
  datastore standby nodes: none
addons:
  enabled:
    ha-cluster           # Configure high availability on the current node
  disabled:
    dashboard            # The Kubernetes dashboard
    dns                  # CoreDNS
    helm                 # Helm 2 - the package manager for Kubernetes
    helm3                # Helm 3 - Kubernetes package manager
    host-access          # Allow Pods connecting to Host services smoothly
    ingress              # Ingress controller for external access
    metallb              # Loadbalancer for your Kubernetes cluster
    metrics-server       # K8s Metrics Server for API access to service metrics
    rbac                 # Role-Based Access Control for authorisation
    registry             # Private image registry exposed on localhost:32000
    storage              # Storage class; allocates storage from host directory
```

node1: `microk8s add-node`
node2: `microk8s join 172.16.10.23:25000/<x>`
node1: `microk8s add-node` (before, or after node2 is finished adding shows the same behaviour)
node3: `microk8s join 172.16.10.23:25000/<y>` (again, before, or after node2 is finished adding shows the same behaviour)

node1, node2, node3: -
```
microk8s status
microk8s is running
high-availability: yes
  datastore master nodes: 172.16.10.23:19001 172.16.10.21:19001 172.16.10.22:19001
  datastore standby nodes: none
addons:
  enabled:
    ha-cluster           # Configure high availability on the current node
  disabled:
    dashboard            # The Kubernetes dashboard
    dns                  # CoreDNS
    helm                 # Helm 2 - the package manager for Kubernetes
    helm3                # Helm 3 - Kubernetes package manager
    host-access          # Allow Pods connecting to Host services smoothly
    ingress              # Ingress controller for external access
    metallb              # Loadbalancer for your Kubernetes cluster
    metrics-server       # K8s Metrics Server for API access to service metrics
    rbac                 # Role-Based Access Control for authorisation
    registry             # Private image registry exposed on localhost:32000
    storage              # Storage class; allocates storage from host directory
```

No calico nodes enter healthy state, stuck in CrashLoopBackoff -
```
microk8s kubectl get all -A
NAMESPACE     NAME                                          READY   STATUS              RESTARTS   AGE
kube-system   pod/calico-kube-controllers-847c8c99d-nv86p   0/1     ContainerCreating   0          14m
kube-system   pod/calico-node-pn6wj                         0/1     CrashLoopBackOff    6          11m
kube-system   pod/calico-node-9sfmr                         0/1     CrashLoopBackOff    6          10m
kube-system   pod/calico-node-2l7pv                         0/1     Running             0          19s

NAMESPACE   NAME                 TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
default     service/kubernetes   ClusterIP   10.152.183.1   <none>        443/TCP   14m

NAMESPACE     NAME                         DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
kube-system   daemonset.apps/calico-node   3         3         0       3            0           kubernetes.io/os=linux   14m

NAMESPACE     NAME                                      READY   UP-TO-DATE   AVAILABLE   AGE
kube-system   deployment.apps/calico-kube-controllers   0/1     1            0           14m

NAMESPACE     NAME                                                DESIRED   CURRENT   READY   AGE
kube-system   replicaset.apps/calico-kube-controllers-847c8c99d   1         1         0       14m
```

```
microk8s kubectl logs -f pod/calico-node-2l7pv -n kube-system
2020-09-08 12:17:16.514 [INFO][9] startup.go 290: Early log level set to info
2020-09-08 12:17:16.514 [INFO][9] startup.go 306: Using NODENAME environment for node name
2020-09-08 12:17:16.514 [INFO][9] startup.go 318: Determined node name: melchior.gaia
2020-09-08 12:17:16.518 [INFO][9] startup.go 350: Checking datastore connection
2020-09-08 12:17:16.566 [INFO][9] startup.go 365: Hit error connecting to datastore - retry error=Get https://10.152.183.1:443/api/v1/nodes/foo: dial tcp 10.152.183.1:443: connect: no route to host
2020-09-08 12:17:17.620 [INFO][9] startup.go 365: Hit error connecting to datastore - retry error=Get https://10.152.183.1:443/api/v1/nodes/foo: dial tcp 10.152.183.1:443: connect: no route to host
2020-09-08 12:17:18.667 [INFO][9] startup.go 365: Hit error connecting to datastore - retry error=Get https://10.152.183.1:443/api/v1/nodes/foo: dial tcp 10.152.183.1:443: connect: no route to host
2020-09-08 12:17:19.705 [INFO][9] startup.go 365: Hit error connecting to datastore - retry error=Get https://10.152.183.1:443/api/v1/nodes/foo: dial tcp 10.152.183.1:443: connect: no route to host
2020-09-08 12:17:20.756 [INFO][9] startup.go 365: Hit error connecting to datastore - retry error=Get https://10.152.183.1:443/api/v1/nodes/foo: dial tcp 10.152.183.1:443: connect: no route to host
2020-09-08 12:17:21.806 [INFO][9] startup.go 365: Hit error connecting to datastore - retry error=Get https://10.152.183.1:443/api/v1/nodes/foo: dial tcp 10.152.183.1:443: connect: no route to host
2020-09-08 12:17:22.858 [INFO][9] startup.go 365: Hit error connecting to datastore - retry error=Get https://10.152.183.1:443/api/v1/nodes/foo: dial tcp 10.152.183.1:443: connect: no route to host
2020-09-08 12:17:23.958 [INFO][9] startup.go 365: Hit error connecting to datastore - retry error=Get https://10.152.183.1:443/api/v1/nodes/foo: dial tcp 10.152.183.1:443: connect: no route to host


Name:                 calico-node-9sfmr                                                                                       
Namespace:            kube-system                                                                                             
Priority:             2000001000                                                                                                                                                                                  
Priority Class Name:  system-node-critical                                                                                                                   
Node:                 balthasar.gaia/172.16.10.22                                                                                                                                                                                                                                                                          
Start Time:           Tue, 08 Sep 2020 12:05:55 +0000                                                                                                        
Labels:               controller-revision-hash=679986bc67                                                                                                    
                      k8s-app=calico-node                                                                                                                    
                      pod-template-generation=2                                                                                                              
Annotations:          scheduler.alpha.kubernetes.io/critical-pod:                                                                                            
Status:               Running                                                                                                                                                                                                                                                                                              
IP:                   172.16.10.22                                                                                                                                                                                                                                                                                         
IPs:                                                                                                                                                         
  IP:           172.16.10.22                                                                                                                                 
Controlled By:  DaemonSet/calico-node                                                                                                                        
Init Containers:                                                                                                                                             
  upgrade-ipam:                                                                                                                                                                                                                                                                                                            
    Container ID:  containerd://f552f728a11b2841aa6b2db28d26eed10c61808a1da5a8e85e748a7b831ea346                                                                                                                                                                                                                           
    Image:         calico/cni:v3.13.2                                                                                                                        
    Image ID:      docker.io/calico/cni@sha256:bbf7e3ac3f80d0a356a6c27b095bd313d1106f8ed84f85850816ed79295843c1                                                                                                                                                                                                            
    Port:          <none>                                                                                                                                                                                         
    Host Port:     <none>                                                                                                                                                                                                                                                                                                  
    Command:                                                                                                                                                                                                      
      /opt/cni/bin/calico-ipam                                                                                                                               
      -upgrade                                                                                                                                               
    State:          Terminated                                                                                                                               
      Reason:       Completed                                                                                                                                
      Exit Code:    0                                                                                                                                        
      Started:      Tue, 08 Sep 2020 12:05:56 +0000                                                                                                          
      Finished:     Tue, 08 Sep 2020 12:05:56 +0000                                                                                                          
    Ready:          True                                                                                                                                     
    Restart Count:  0                                                                                                                                        
    Environment:                                                                                                                                             
      KUBERNETES_NODE_NAME:        (v1:spec.nodeName)                                                                                                        
      CALICO_NETWORKING_BACKEND:  <set to the key 'calico_backend' of config map 'calico-config'>  Optional: false                                                                                                                                                                                                         
    Mounts:                                                                                                                                                  
      /host/opt/cni/bin from cni-bin-dir (rw)                                                                                                                                                                     
      /var/lib/cni/networks from host-local-net-dir (rw)                                                                                                                                                          
      /var/run/secrets/kubernetes.io/serviceaccount from calico-node-token-l6vc2 (ro)                                                                                                                             
  install-cni:                                                                                                                                                                                                    
    Container ID:  containerd://65e3e9bf7133b22ef8c4c116ff00cdf5056337c76426accdab9ed0b48612dce2                                                                                                                  
    Image:         calico/cni:v3.13.2                                                                                                                                                                             
    Image ID:      docker.io/calico/cni@sha256:bbf7e3ac3f80d0a356a6c27b095bd313d1106f8ed84f85850816ed79295843c1                                                                                                   
    Port:          <none>                                                                                                                                                                                         
    Host Port:     <none>                                                                                                                                                                                         
    Command:                                                                                                                                                                                                      
      /install-cni.sh                                                                                                                                                                                             
    State:          Terminated                                                                                                                                                                                    
      Reason:       Completed                                                                                                                                                                                     
      Exit Code:    0                                                                                                                                                                                             
      Started:      Tue, 08 Sep 2020 12:05:57 +0000                                                                                                                                                               
      Finished:     Tue, 08 Sep 2020 12:05:58 +0000                                                                                                                                                               
    Ready:          True                                                                                                                                                                                          
    Restart Count:  0                                                                                                                                                                                             
    Environment:                                                                                                                                                                                                  
      CNI_CONF_NAME:         10-calico.conflist                                                                                                                                                                   
      CNI_NETWORK_CONFIG:    <set to the key 'cni_network_config' of config map 'calico-config'>  Optional: false                                                                                                 
      KUBERNETES_NODE_NAME:   (v1:spec.nodeName)                                                                                                                                                                  
      CNI_MTU:               <set to the key 'veth_mtu' of config map 'calico-config'>  Optional: false                                                                                                           
      SLEEP:                 false                                                                                                                                                                                
      CNI_NET_DIR:           /var/snap/microk8s/current/args/cni-network                                                                                                                                          
    Mounts:                                                                                                                                                                                                       
      /host/etc/cni/net.d from cni-net-dir (rw)                                                                                                                                                                   
      /host/opt/cni/bin from cni-bin-dir (rw)                                                                                                                                                                     
      /var/run/secrets/kubernetes.io/serviceaccount from calico-node-token-l6vc2 (ro)
Containers:                                                                                                                                                                         
  calico-node:                                                                                                                                                                      
    Container ID:   containerd://4461881bfcf4698c394e2f5996836d4eebfea1bf66fdc9427e679e3e51e6ebe5                                                                                   
    Image:          calico/node:v3.13.2                                                                                                                                             
    Image ID:       docker.io/calico/node@sha256:cb9dea7b86471c71925ae318f7c60af72d9ddf1dab0fe2029832a671b83bba6a                                                                   
    Port:           <none>                                                                                                                                                          
    Host Port:      <none>                                                                                                                                                          
    State:          Running                                                                                                                                                         
      Started:      Tue, 08 Sep 2020 12:18:19 +0000                                                                                                                                 
    Last State:     Terminated                                                                                                                                                      
      Reason:       Error                                                                                                                                                           
      Exit Code:    137                                                                                                                                                                                                                                                                                                                                                 
      Started:      Tue, 08 Sep 2020 12:14:18 +0000                                                                                                                                 
      Finished:     Tue, 08 Sep 2020 12:15:27 +0000                                                                                                                                                                                                                                                                                                                     
    Ready:          False                                                                                                                                                           
    Restart Count:  7                                                                                                                                                                                                                                                                                                      
    Requests:                                                                                                                                                                                                                                                                                                              
      cpu:      250m                                                                                                                                                                                                                                                                                                       
    Liveness:   exec [/bin/calico-node -felix-live] delay=10s timeout=1s period=10s #success=1 #failure=6                                                                                                         
    Readiness:  exec [/bin/calico-node -felix-ready] delay=0s timeout=1s period=10s #success=1 #failure=3                                                                                                                                                                                                                  
    Environment:                                                                                                                                                                                                  
      DATASTORE_TYPE:                     kubernetes                                                                                                                                                                                                                                                                       
      WAIT_FOR_DATASTORE:                 true                                                                                                                                                                                                                                                                             
      NODENAME:                            (v1:spec.nodeName)                                                                                                                                                                                                                                                              
      CALICO_NETWORKING_BACKEND:          <set to the key 'calico_backend' of config map 'calico-config'>  Optional: false                                                                                        
      CLUSTER_TYPE:                       k8s,bgp                                                                             
      IP:                                 autodetect                                                                          
      IP_AUTODETECTION_METHOD:            can-reach=172.16.10.21                                                              
      CALICO_IPV4POOL_VXLAN:              Always                                                                              
      FELIX_IPINIPMTU:                    <set to the key 'veth_mtu' of config map 'calico-config'>  Optional: false                                                                                              
      CALICO_IPV4POOL_CIDR:               10.1.0.0/16                                                                                                                                                                                                                                                                      
      CALICO_DISABLE_FILE_LOGGING:        true                                                                                
      FELIX_DEFAULTENDPOINTTOHOSTACTION:  ACCEPT                                                                                                                                                                                                                                                                           
      FELIX_IPV6SUPPORT:                  false                                                                                                                                                                                                                                                                            
      FELIX_LOGSEVERITYSCREEN:            error                                                                                                                                                                   
      FELIX_HEALTHENABLED:                true                                                                                                                                                                                                                                                                             
    Mounts:                                                                                                                                                                                                                                                                                                                
      /lib/modules from lib-modules (ro)                                                                                                                                                                                                                                                                                   
      /run/xtables.lock from xtables-lock (rw)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
      /var/lib/calico from var-lib-calico (rw)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
      /var/run/calico from var-run-calico (rw)                                                                                
      /var/run/nodeagent from policysync (rw)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
      /var/run/secrets/kubernetes.io/serviceaccount from calico-node-token-l6vc2 (ro)                                         
Conditions:                                                                                                                   
  Type              Status                                                                                                                                                                                        
  Initialized       True                                                                                                                                     
  Ready             False                                                                                                                                                                                                                                                                                                  
  ContainersReady   False                                                                                                                                    
  PodScheduled      True                                                                                                                                     
Volumes:                                                                                                                                                     
  lib-modules:                                                                                                                                               
    Type:          HostPath (bare host directory volume)                                                                                                     
    Path:          /lib/modules                                                                                                                                                                                                                                                                                            
    HostPathType:                                                                                                                                                                                                                                                                                                          
  var-run-calico:                                                                                                                                            
    Type:          HostPath (bare host directory volume)                                                                                                     
    Path:          /var/snap/microk8s/current/var/run/calico                                                                                                 
    HostPathType:                                                                                                                                            
  var-lib-calico:                                                                                                                                                                                                                                                                                                          
    Type:          HostPath (bare host directory volume)                                                                                                                                                                                                                                                                   
    Path:          /var/snap/microk8s/current/var/lib/calico                                                                                                 
    HostPathType:                                                                                                                                                                                                                                                                                                          
  xtables-lock:                                                                                                                                                                                                   
    Type:          HostPath (bare host directory volume)                                                                                                                                                                                                                                                                   
    Path:          /run/xtables.lock                                                                                                                                                                              
    HostPathType:  FileOrCreate                                                                                                                              
  cni-bin-dir:                                                                                                                                               
    Type:          HostPath (bare host directory volume)                                                                                                     
    Path:          /var/snap/microk8s/current/opt/cni/bin                                                                                                    
    HostPathType:                                                                                                                                            
  cni-net-dir:                                                                                                                                               
    Type:          HostPath (bare host directory volume)                                                                                                     
    Path:          /var/snap/microk8s/current/args/cni-network                                                                                               
    HostPathType:                                                                                                                                            
  host-local-net-dir:                                                                                                                                        
    Type:          HostPath (bare host directory volume)                                                                                                     
    Path:          /var/snap/microk8s/current/var/lib/cni/networks                                                                                                                                                                                                                                                         
    HostPathType:                                                                                                                                            
  policysync:                                                                                                                                                                                                     
    Type:          HostPath (bare host directory volume)                                                                                                                                                          
    Path:          /var/snap/microk8s/current/var/run/nodeagent                                                                                                                                                   
    HostPathType:  DirectoryOrCreate                                                                                                                                                                              
  flexvol-driver-host:                                                                                                                                                                                            
    Type:          HostPath (bare host directory volume)                                                                                                                                                          
    Path:          /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds                                                                                                                              
    HostPathType:  DirectoryOrCreate                                                                                                                                                                              
  calico-node-token-l6vc2:                                                                                                                                                                                        
    Type:        Secret (a volume populated by a Secret)                                                                                                                                                          
    SecretName:  calico-node-token-l6vc2                                                                                                                                                                          
    Optional:    false                                                                                                                                                                                            
QoS Class:       Burstable                                                                                                                                                                                        
Node-Selectors:  kubernetes.io/os=linux                                                                                                                                                                           
Tolerations:     :NoScheduleop=Exists                                                                                                                                                                             
                 :NoExecuteop=Exists                                                                                                                                                                              
                 CriticalAddonsOnly op=Exists                                                                                                                                                                     
                 node.kubernetes.io/disk-pressure:NoSchedule op=Exists                                                                                                                                            
                 node.kubernetes.io/memory-pressure:NoSchedule op=Exists                                                                                                                                          
                 node.kubernetes.io/network-unavailable:NoSchedule op=Exists                                                                                                                                      
                 node.kubernetes.io/not-ready:NoExecute op=Exists                                                                                                                                                 
                 node.kubernetes.io/pid-pressure:NoSchedule op=Exists                                                                                                                                             
                 node.kubernetes.io/unreachable:NoExecute op=Exists                                                                                                                                               
                 node.kubernetes.io/unschedulable:NoSchedule op=Exists                                                                                                                                            
Events:                                                                                                                                                                                                           
  Type     Reason     Age                     From                     Message                                                                                                                                    
  ----     ------     ----                    ----                     -------                                                                                                                                    
  Normal   Scheduled  12m                                              Successfully assigned kube-system/calico-node-9sfmr to balthasar.gaia                                                                      
  Normal   Pulled     12m                     kubelet, balthasar.gaia  Container image "calico/cni:v3.13.2" already present on machine                                                                            
  Normal   Created    12m                     kubelet, balthasar.gaia  Created container upgrade-ipam                                                                                                             
  Normal   Started    12m                     kubelet, balthasar.gaia  Started container upgrade-ipam                                                                                                             
  Normal   Pulled     12m                     kubelet, balthasar.gaia  Container image "calico/cni:v3.13.2" already present on machine                                                                            
  Normal   Created    12m                     kubelet, balthasar.gaia  Created container install-cni                                                                                                              
  Normal   Started    12m                     kubelet, balthasar.gaia  Started container install-cni                                                                                                              
  Normal   Pulled     12m                     kubelet, balthasar.gaia  Container image "calico/pod2daemon-flexvol:v3.13.2" already present on machine                                                             
  Normal   Created    12m                     kubelet, balthasar.gaia  Created container flexvol-driver                                                                                                           
  Normal   Started    12m                     kubelet, balthasar.gaia  Started container flexvol-driver                                                                                                           
  Normal   Pulled     12m                     kubelet, balthasar.gaia  Container image "calico/node:v3.13.2" already present on machine                                                                           
  Normal   Created    12m                     kubelet, balthasar.gaia  Created container calico-node                                                                                                              
  Normal   Started    12m                     kubelet, balthasar.gaia  Started container calico-node                                                                                                              
  Warning  Unhealthy  11m (x6 over 12m)       kubelet, balthasar.gaia  Liveness probe failed: calico/node is not ready: Felix is not live: Get http://localhost:9099/liveness: dial tcp [::1]:9099: connect: connection refused                                                                                                                                                                                                     
  Normal   Killing    11m                     kubelet, balthasar.gaia  Container calico-node failed liveness probe, will be restarted                                                                             
  Warning  Unhealthy  7m51s (x30 over 12m)    kubelet, balthasar.gaia  Readiness probe failed: calico/node is not ready: felix is not ready: Get http://localhost:9099/readiness: dial tcp [::1]:9099: connect: connection refused                                                                                                                                                                                                  
  Warning  BackOff    2m43s (x12 over 5m54s)  kubelet, balthasar.gaia  Back-off restarting failed container 
```

**Comment by devec0** [comment]
[inspection-report-20200908_122206.tar.gz](https://github.com/ubuntu/microk8s/files/5188317/inspection-report-20200908_122206.tar.gz)

**Comment by balchua** [comment]
It can be related to this issue https://github.com/ubuntu/microk8s/issues/1511

Care to try `latest/edge` channel, there was a fix to address the that issue.

**Comment by balchua** [comment]
Hi @devec0  this maybe something introduced in 1.19.
I can see that it seems to be trying to use ipv6.

See this issue #1549 and how it was resolved. Is it something that you can try?

**Comment by ktsakalozos** [comment]
Hi @devec0,

What can you tell me about the kernel you use? I see that kube-proxy cannot load a module:
```
Sep 08 12:05:28 balthasar.gaia microk8s.daemon-proxy[1129352]: + modprobe br_netfilter
Sep 08 12:05:28 balthasar.gaia microk8s.daemon-proxy[1129594]: modprobe: FATAL: Module br_netfilter not found in directory /lib/modules/5.6.0-1137-ayufan-ge57f05e7bf8f
``` 
Could you make sure this module is loaded?

**Comment by devec0** [comment]
I did end up trying the latest edge channel, and got the same result.
I indeed suspect this could be kernel related. I'd not noticed the br_netfilter message, and the more I read, the more that module seems to be required by several components in K8S. I'm in the process of rebuilding the kernel for this device (RockPro64) to include the standard Ubuntu modules and will report back once tested.

**Comment by devec0** [comment]
OK, so, I think this can be closed out. I do think there is some opportunity to add something to the documentation around required kernel modules and configuration, as I've tried several permutations of different kernel versions and kernel builds using the standard Ubuntu kernel configuration file for ARM64, and have finally found a version and configuration that works. Unsurprisingly, the default Ubuntu kernel configuration file produces an ARM64 kernel on the 5.4.x focal kernels which works with microk8s ha-cluster, on the 1.19 version. Stepping outside of these constraints, and thar be dragons.

In summary, I observed that the issue was resolved by using the `1.19/stable` channel, the standard Ubuntu focal 5.4 kernel from kernel.ubuntu.com git, compiled with the default Ubuntu kernel configuration for ARM64. I can confirm that this configuration does include br_netfilter, including all of the kernel modules listed in the [Calico requirements documentation](https://docs.projectcalico.org/getting-started/kubernetes/requirements). 

As an interesting aside, the HWE-5.8 kernel was tested, but exhibited some strange failure modes on bringing up microk8s ha-clustering, so I decided to go back to the the 5.4 branch and recompile. This is probably worth being aware of for future HWE testing, once that kernel is released, as I noted issues in bringing up the apiserver, which inevitably caused microk8s to fail during ha-cluster bootstrap. This could either be related to the standard HWE-5.8 kernel config or some breaking change in 5.8 - If I see that again, I'll open a separate issue.

As an additional note, but also noting that latest/edge is under active development, the current latest/edge does seem to also have an issue during ha-cluster bringup, even on the working kernel configuration I mentioned (5.4 + standard ARM64 Ubuntu kernel configuration) but it was not related to this issue as far as I can tell. In my testing on latest/edge, I found I had to go back to 1.19 on each kernel I tested, and was only able to get working ha-cluster once I switched back to 1.19/stable. The apiserver would simply fail whilst applying CNI configuration, nothing stood out in the logs. Also, if I hit this again during future upgrades, I'll be sure to file another issue.

For the sake of completeness:
```
microk8s kubectl get all -A
NAMESPACE     NAME                                          READY   STATUS    RESTARTS   AGE
kube-system   pod/calico-node-ztxlv                         1/1     Running   0          14m
kube-system   pod/calico-kube-controllers-847c8c99d-chpkf   1/1     Running   1          16m
kube-system   pod/calico-node-m6rw2                         1/1     Running   0          13m
kube-system   pod/calico-node-jvz9r                         1/1     Running   0          13m

NAMESPACE   NAME                 TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
default     service/kubernetes   ClusterIP   10.152.183.1   <none>        443/TCP   16m

NAMESPACE     NAME                         DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
kube-system   daemonset.apps/calico-node   3         3         3       3            3           kubernetes.io/os=linux   16m

NAMESPACE     NAME                                      READY   UP-TO-DATE   AVAILABLE   AGE
kube-system   deployment.apps/calico-kube-controllers   1/1     1            1           16m

NAMESPACE     NAME                                                DESIRED   CURRENT   READY   AGE
kube-system   replicaset.apps/calico-kube-controllers-847c8c99d   1         1         1       16m
```

```
microk8s status
microk8s is running
high-availability: yes
  datastore master nodes: 172.16.10.21:19001 172.16.10.22:19001 172.16.10.23:19001
  datastore standby nodes: none
addons:
  enabled:
    ha-cluster           # Configure high availability on the current node
  disabled:
    dashboard            # The Kubernetes dashboard
    dns                  # CoreDNS
    helm                 # Helm 2 - the package manager for Kubernetes
    helm3                # Helm 3 - Kubernetes package manager
    host-access          # Allow Pods connecting to Host services smoothly
    ingress              # Ingress controller for external access
    metallb              # Loadbalancer for your Kubernetes cluster
    metrics-server       # K8s Metrics Server for API access to service metrics
    rbac                 # Role-Based Access Control for authorisation
    registry             # Private image registry exposed on localhost:32000
    storage              # Storage class; allocates storage from host directory
```
